### PR TITLE
Patient space functionality

### PR DESCRIPTION
**Pull request by OllieGYoung**
The key following files have been altered;

- endpoints.py
- db.py
- parser.py
- panelapp.py

**Summary of added functions;**
Patient Space functionality added. 
A user can now input a patient ID to retrieve the all the tests that the patient has had.

A user can now also input the patient ID and R code to generate a comparison of panel contents between a historical panel given to a patient and the current panel versions' contents, if such a difference exists.

Currently this feature doesn't update the database if the online panel version differs from the local latest panel version. 

The code still needs further documentation, error handling and logging, however this acts as a functional precursor for development. 

The update patient endpoint has been created, however the backend functionality will be pushed in the next commit.

### CODE REVIEW

**Review by Mani-varma1 — APPROVED** [review]
will raise issues once i have it tested fully

**Review by Mo-shalakani — APPROVED** [review]
neat!

**Review by VIBE-TG — APPROVED** [review]
Good

File: vimmo/API/endpoints.py [hunk 1]
```diff
@@ -99,6 +101,20 @@ def get(self):
...
+                return panel_data
+            gene_query={record["HGNC_ID"] for record in panel_data["Associated Gene Records"]}
+            gene_query="|".join(gene_query)
+        elif r_code:
+            panel_data = query.get_panels_by_rcode(rcode=args.get("Rcode"), matches=args.get("Similar_Matches"))
+            if "Message" in panel_data:
+                return panel_data
+            gene_query={record["HGNC_ID"] for record in panel_data["Associated Gene Records"]}
+            gene_query="|".join(gene_query)
         query = PanelQuery(db.conn)
```
Comment (OllieGYoung) [inline vimmo/API/endpoints.py:118]: Syntax err

Comment (Mani-varma1) [reply vimmo/API/endpoints.py]: i think i changed this so this is handled in a different place. Not sure will test it

**Review by Itebbs22 — APPROVED** [review]

**Comment by Mani-varma1** [comment]
I am going to merge so i can work on this tomorrow

**Pull request merged**